### ISSUE TITLE
[Docs] Clarify setup.cfg-only Package Release Tagging

### ISSUE THREAD

**Issue by markgras** [pypa/setuptools#2819]
### Summary

Reading the current setuptools v58.2.0 documentation, it is unclear to me how to tag setup.cfg-only package releases. [This page](https://setuptools.pypa.io/en/latest/userguide/distribution.html#making-official-non-snapshot-releases) mentions that you can add tag settings to your setup.cfg file. However, [this page](https://setuptools.pypa.io/en/latest/userguide/declarative_config.html) does not mention this at all.

My specific goal in reading the documentation was to learn how to tag a setup.cfg-only package with a custom build tag, in the style of `--tag-build=STRING`.

### Additional Information

This documentation change better helps developers move toward a setup.cfg-only setuptools future.

### Code of Conduct

- [X] I agree to follow the PSF Code of Conduct

**Comment by abravalheri** [comment]
Hi @markgras, thank you very much for opening this issue.

Any chance this is related to https://github.com/pypa/setuptools/issues/2491?

**Comment by markgras** [comment]
Not necessarily. This issue is that the documentation doesn't tell a user how to tag setup.cfg-only packages. If passing tags to `python -m build` is supposed to work and is the intended way that users are to tag setup.cfg-only packages, then #2491 is related. I did at one point try passing tags to `python -m build` and it didn't work, but it was unclear to me if I was supposed to be doing that.

**Comment by abravalheri** [comment]
Thank you very much for the clarification @markgras!
Yes, indeed this topic needs better docs, however it is likely that it will take a bit longer because there are some discussions going on.

Meanwhile, if you are using `build`, there is a hint on how this can be done in https://github.com/pypa/build/issues/202#issuecomment-757375626, but I suspect you can also try borrowing a bit of setuptools' own configuration:

https://github.com/pypa/setuptools/blob/6092cdcbd659110e613e9841cdd5a0c43e61f33b/setup.cfg#L138-L140

This is a [`distutils` configuration](https://docs.python.org/3/distutils/configfile.html), [but there might be some dragons](https://github.com/pypa/setuptools/issues/2529).

**Comment by markgras** [comment]
Understood. Thank you for sharing these workarounds.
